feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { AsmBindCartFacade } from '@spartacus/asm/root';
+import { ActiveCartFacade, MultiCartFacade } from '@spartacus/cart/base/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  HttpErrorModel,
+  OCC_CART_ID_CURRENT,
+} from '@spartacus/core';
+import { BehaviorSubject, of, Subscription } from 'rxjs';
+import { concatMap, filter, finalize, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-asm-bind-cart',
+  templateUrl: './asm-bind-cart.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AsmBindCartComponent implements OnInit, OnDestroy {
```

Comment:
Just for the sake of discussion, the use of the ASM for classes and interfaces started back when ASM was not in its own feature library.  It is perhaps less useful now and a bit redindant.  I'm not insisting on a change of name for the new class.  I just want to point it out.